components/service/glean/src/main/java/mozilla/components/service/glean/Dispatchers.kt

```diff
@@ -34,6 +34,12 @@ internal object Dispatchers {
                     block()
                 }
                 null
+            } else if (!Glean.isInitialized()) {
```

Comment:
I like this approach! However, I feel like this should not depend on the `Glean`. Instead, we could:

- have the list of queued coroutines as an implementation detail of the `Dispatchers` object;
- provide a flag for changing the behaviour (as we do for `testingMode`; this would also make sure we don't need to flip the behaviour of other Glean tests as we could only flip the behaviour on for a specific test;
- control the Dispatcher's behaviour *within* `Glean.initialize` (so that the init logic is only in one place)
- we should somehow make sure this is thread-safe (enqueuing and dispatching the queue should not stumble upon each other);